src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/WebServicesCmdletBase.cs

```diff
@@ -0,0 +1,261 @@
+using System.Threading;
+using Microsoft.Azure.Commands.MachineLearning.Extensions;
+using Microsoft.Azure.Commands.MachineLearning.Utilities;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Entities.ErrorResponses;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Extensions;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.Rest.Azure;
+using PSResourceManagerModels = Microsoft.Azure.Commands.Resources.Models;
+
+namespace Microsoft.Azure.Commands.MachineLearning
+{
+    public abstract class WebServicesCmdletBase : AzureRMCmdlet
+    {
+        public const string CommandletSuffix = "AzureRmMlWebService";
+
+        private PSResourceManagerModels.ResourcesClient resourcesClient;
+
+        private WebServicesClient webServicesClient;
+
+        /// <summary>
+        /// The cancellation source.
+        /// </summary>
+        private CancellationTokenSource cancellationSource;
+        
+        protected CancellationToken? CancellationToken
+        {
+            get
+            {
+                return this.cancellationSource == null ? null : (CancellationToken?)this.cancellationSource.Token;
+            }
+        }
+
+        protected string SubscriptionId { get; set; }
+
+        public PSResourceManagerModels.ResourcesClient ResourceClient
+        {
+            get
+            {
+                if (this.resourcesClient == null)
+                {
+                    this.resourcesClient = new PSResourceManagerModels.ResourcesClient(DefaultProfile.Context)
+                    {
+                        VerboseLogger = WriteVerboseWithTimestamp,
+                        ErrorLogger = WriteErrorWithTimestamp,
+                        WarningLogger = WriteWarningWithTimestamp
+                    };
+                }
+                return this.resourcesClient;
+            }
+            set { this.resourcesClient = value; }
+        }
+        
+        public WebServicesClient WebServicesClient
+        {
+            get
+            {
+                if (this.webServicesClient == null)
+                {
+                    this.webServicesClient = new WebServicesClient(DefaultProfile.Context)
+                    {
+                        VerboseLogger = WriteVerboseWithTimestamp,
+                        ErrorLogger = WriteErrorWithTimestamp,
+                        WarningLogger = WriteWarningWithTimestamp
+                    };
+                }
+                return this.webServicesClient;
+            }
+            set { this.webServicesClient = value; }
+        }
+
+        #region Processing life cycle
+
+        protected override void BeginProcessing()
+        {
+            try
+            {
+                if (this.cancellationSource == null)
+                {
+                    this.cancellationSource = new CancellationTokenSource();
+                }
+
+                this.SubscriptionId = this.DefaultProfile.Context.Subscription.Id.ToString();
```

Comment:
Where is it used?  The subscription id is aready injected in the client when you create it, and is always accessbile from the context, if needed (cmdlets should not need it).